### ISSUE TITLE
[FORMATTING] Column called "type" in Snowflake

### ISSUE THREAD

**Issue by greg-finley** [sql-formatter-org/sql-formatter#771]
**Input data**

Which SQL and options did you provide as input?

```sql
 SELECT 
        SUM(CASE WHEN type = 'upgrade' THEN amount ELSE 0 END) AS upgrade_revenue
    FROM adjustments
```
Default settings, except dialect Snowflake

**Expected Output**

No parse error

**Actual Output**

```
An Unexpected Error Occurred
Parse error at token: TYPE at line 3 column 23 Unexpected RESERVED_CLAUSE token:
```

**Usage**

- How are you calling / using the library? From our app and from your online demo tool
- What SQL language(s) does this apply to? Snowflake
- Which SQL Formatter version are you using? Latest

Changing the variable name to something else like `my_type` instead of `type` will parse correctly.

**Comment by nene** [comment]
Thanks for reporting. Looks like this is the same issue that was fixed for PostgreSQL: #685

**Comment by nene** [comment]
Fixed in 15.4.1 release
